pipeline/e2e-test.yaml

```diff
@@ -5,5 +5,24 @@ steps:
     - script: yarn build:all
       displayName: build:all
 
+    - script: docker build -t app .
+      condition: eq( variables['Agent.OS'], 'Linux' )
```

Comment:
Would it be more clear about why the environment checks are happening to make the "docker or not" decision part of the strategy matrix, rather than adding the environment check to each individual task? ie, in strategy,

```
      strategy:
          matrix:
              windows:
                  imageName: 'vs2017-win2016'
                  e2eEnvironment: 'agent'
              linux:
                  imageName: 'ubuntu-16.04'
                  e2eEnvironment: 'docker'
```

...and then in the tasks, `condition: eq( variables['e2eEnvrionment'], 'docker' )`/etc?

Reply:
I have removed the `strategy` and `environment checks` in favor of a separate `ubuntu` only job that uses a `docker` yaml file. So we dont need all of those checks anymore.